src/Tools/dotnet-monitor/Program.cs

```diff
@@ -38,10 +39,29 @@ private static Command CollectCommand()
                 name: "collect",
                 description: Strings.HelpDescription_CommandCollect)
             {
-                SharedOptions()
+                UrlsOption,
+                MetricUrlsOption,
+                ProvideMetricsOption,
+                DiagnosticPortOption,
+                NoAuthOption,
+                TempApiKeyOption,
+                NoHttpEgressOption,
+                ConfigurationFilePathOption
             };
 
-            command.SetHandler<CancellationToken, string[], string[], bool, string, bool, bool, bool, FileInfo>(CollectCommandHandler.Invoke, command.Children.OfType<IValueDescriptor>().ToArray());
+            command.SetHandler(async context =>
+            {
+                context.ExitCode = await CollectCommandHandler.Invoke(
+                    context.GetCancellationToken(),
+                    context.ParseResult.GetValueForOption(UrlsOption),
```

Comment:
Is this version of the Invoke command strongly typed? I.e. if there's a mismatch between the children declared above and the results passed here, what happens?

Reply:
Yes, the Inovke method is something that we wrote. In this change, we are manually getting each corresponding value from each option and passing it into the Invoke method. If someone adds a new parameter to the Invoke method or changes the type of an existing parameter, this code will fail to compile, and it should be obvious why.

Reply:
If another option is added to the Command instance but not consumed by the Invoke invocation, then that would not be obvious why that is. However, I would expect that someone who is adding a new option also attempted to plumb through the use of the new information somewhere and that would have had to been surfaced up at the Invoke method.